tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -400,6 +400,61 @@ var _ = Describe("odo devfile push command tests", func() {
 		})
 	})
 
+	Context("Verify devfile volume components work", func() {
+
+		It("should error out when duplicate volume components exist", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.RenameFile("devfile.yaml", "devfile-old.yaml")
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-volume-components.yaml"), filepath.Join(context, "devfile.yaml"))
+
+			helper.ReplaceString("devfile.yaml", "secondvol", "firstvol")
+
+			output := helper.CmdShouldFail("odo", "push", "--project", namespace)
+			Expect(output).To(ContainSubstring("duplicate volume components present in devfile"))
+		})
+
+		It("should error out when a wrong volume size is used", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.RenameFile("devfile.yaml", "devfile-old.yaml")
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-volume-components.yaml"), filepath.Join(context, "devfile.yaml"))
+
+			helper.ReplaceString("devfile.yaml", "3Gi", "3Garbage")
+
+			output := helper.CmdShouldFail("odo", "push", "--project", namespace)
+			Expect(output).To(ContainSubstring("quantities must match the regular expression"))
+		})
+
+		It("should successfully use the volume components in container components", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.RenameFile("devfile.yaml", "devfile-old.yaml")
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-volume-components.yaml"), filepath.Join(context, "devfile.yaml"))
+
+			output := helper.CmdShouldPass("odo", "push", "--project", namespace)
+			Expect(output).To(ContainSubstring("Changes successfully pushed to component"))
+
+			// Only testing this in Kubernetes because odo OpenShift CI
+			// sets the min pvc storage size to 100Gi
+			if os.Getenv("KUBERNETES") == "true" {
```

Comment:
Can we add this condition in JustAfterEach()

Reply:
Not sure what the point is in `JustAfterEach`? in `AfterEach` we remove the namespace/project. So the pvc will be gone by then.